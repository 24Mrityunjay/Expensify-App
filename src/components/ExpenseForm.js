import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/initialize';

export default class ExpenseForm extends React.Component{
    constructor(props){
        super(props);
        this. state = {
            description: props.expense ? props.expense.description : "",
            note: props.expense ? props.expense.note : "",
            createdAt:props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused:false,
            error:'',
            amount: props.expense ? (props.expense.amount).toString() : ""
        };
    }
      
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(()=>({description}));
    };
    onTextAreaChange = (e) => {
        e.persist();
        this.setState(()=>({note : e.target.value}))
    };
    onAmountChange = (e) => {
        const amount = e.target.value;
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
            this.setState(()=>({amount}));
        }
    };
    onDateChange = (createdAt) => {
        this.setState(() => ({ createdAt }));
    };
    onFocusChange = ({focused}) => {
        this.setState(()=>({calendarFocused: focused}));
    };
    onSubmit = (e) => {
        e.preventDefault();
        if(!this.state.description || !this.state.amount){
            this.setState(()=>({
                error:"Please provide description and amount"
            }));
            //alert("Please provide description and amount");
        }else{
            this.setState(()=>({
                error:""
            }));
            this.props.onSubmit({
                description:this.state.description,
                amount:parseFloat(this.state.amount, 10) ,
                note:this.state.note,
                createdAt:this.state.createdAt.valueOf()
            });
            //remove error
            console.log("Form Submitted Successfully!");
            // this.setState(()=>({
            //     description:"",
            //     note:'',
            //     createdAt:moment(),
            //     calendarFocused:false,
            //     error:'',
            //     amount:''
            // }));
        }
    };
    render(){
        return (
            <div>
                <h1>ExpenseForm</h1>
                {this.state.error && <span style={{color: "red"}}>{this.state.error}</span>}
                <form onSubmit={this.onSubmit}>
                 <div>   
                     <input 
                        type="text"
                        placeholder="AddExpense"
                        value={this.state.description}
                        onChange={this.onDescriptionChange}                        
                        autoFocus
                     />
                     <input 
                        type="number"
                        placeholder="Enter Amount"
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                     /><br/>
                     
                 </div>
                 <div>
                     <textarea
                        placeholder="Enter a note for your Expense(optional)"
                        value={this.state.note}
                        onChange={this.onTextAreaChange}
                     ></textarea>
                 </div>
                 <div>
                     <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths= {1}
                        isOutsideRange={()=>false}
                     /><br/>   
                       
                 </div>          
                    <button>Add Expense</button>
                </form>  
            </div>
        )
    }
}