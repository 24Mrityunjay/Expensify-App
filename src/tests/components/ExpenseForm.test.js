import React from 'react';
import { shallow, mount } from 'enzyme';
import moment from 'moment';
import ReactShallowRenderer  from "react-test-renderer/shallow";
import ExpenseForm from '../../components/ExpenseForm';



const expenses = [{    
    id: '2',
    description: 'Coffee',
    note: '', 
    amount: 9500,
    createdAt: 400
}, {
    id: '2',
    description: 'Rent',
    note: '', 
    amount: 109500,
    createdAt: 1000
}, {
    id: '3',
    description: 'Credit Card',
    note: '', 
    amount: 4500,
    createdAt: -1000
}];

test('Should refer expensesList without expenses', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<ExpenseForm />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('Should refer expensesList with expenses', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<ExpenseForm expense = {expenses[0]}/>);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test(' error for invalid form submission', () => {
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<ExpenseForm expense = {expenses[0]}/>);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();

    const wrapper = shallow(<ExpenseForm/>);
    wrapper.find('form').simulate('submit', {preventDefault:()=>{}});
    expect(wrapper.state('error').length).toBeGreaterThan(0);
    expect(wrapper.debug()).toMatchSnapshot();
});

test(' Should set description on input change', () => {
    const value = 'New Description';
    const wrapper = shallow(<ExpenseForm/>);
    wrapper.find('input').at(0).simulate('change',{
        target: {value}
    });
    expect(wrapper.state('description')).toBe(value);
    expect(wrapper.debug()).toMatchSnapshot();
});

test(' Should set valid amount on input change', () => {
    const value = '23.23';
    const wrapper = shallow(<ExpenseForm/>);
    wrapper.find('input').at(1).simulate('change',{target:{value}});
    expect(wrapper.state('amount')).toBe(value);
    expect(wrapper.debug()).toMatchSnapshot();
});

test('should not display if amount is invalid on input change', () => {
    const value = '23.223';
    const wrapper = shallow(<ExpenseForm/>);
    wrapper.find('input').at(1).simulate('change',{target:{value}});
    //expect(wrapper.state('amount')).toBe('');
    expect(wrapper.debug()).toMatchSnapshot();
});

test(' Should set note on textarea change', () => {
    const value = 'note';
    const wrapper = shallow(<ExpenseForm/>);
    wrapper.find('textarea').simulate('change',{target:{value},persist:()=>{}});
    expect(wrapper.state('note')).toBe(value);
    expect(wrapper.debug()).toMatchSnapshot();
});

//SPIES 

test("Should test submit with error and with complete info", () => {
    const onSubmitSpy = jest.fn();
    const wrapper = shallow(<ExpenseForm expense={expenses[0]} onSubmit={onSubmitSpy}/>);
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });
    expect(wrapper.state('error')).toBe('');
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
        description:expenses[0].description,
        amount:expenses[0].amount,
        note:expenses[0].note,
        createdAt:expenses[0].createdAt
    });
    expect(wrapper.debug()).toMatchSnapshot();
});

// test('Should render date change ondatechange', () => {
//     const now = moment();
//     const wrapper = shallow(<ExpenseForm />);
//     wrapper.find('SingleDatePicker').props('onDateChange')({now});
//     expect(wrapper.state('createdAt').debug()).toEqual(now);
//     expect(wrapper.debug()).toMatchSnapshot();
// });

test('Should render calendar change onCalendarfocused', () => {
    const focused = true;
    const wrapper = shallow(<ExpenseForm />);
    wrapper.find('SingleDatePicker').props('onFocusChange')({focused});
    expect(wrapper.state('calendarFocused').debug()).toEqual(focused);
    expect(wrapper.debug()).toMatchSnapshot();
});


// test('Should render calendar change onCalendarfocused', () => {
//     const focused = true;
//     const wrapper = shallow(<ExpenseForm />);
//     wrapper.find('SingleDatePicker').prop('onFocusChange')({focused});
//     expect(wrapper.state('calendarFocused').debug()).toEqual(focused);
//     expect(wrapper.debug()).toMatchSnapshot();
// });