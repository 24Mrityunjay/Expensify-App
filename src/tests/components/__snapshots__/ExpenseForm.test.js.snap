// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Should set description on input change 1`] = `
"<div>
  <h1>
    ExpenseForm
  </h1>
  <form onSubmit={[Function]}>
    <div>
      <input type=\\"text\\" placeholder=\\"AddExpense\\" value=\\"New Description\\" onChange={[Function]} autoFocus={true} />
      <input type=\\"number\\" placeholder=\\"Enter Amount\\" value=\\"\\" onChange={[Function]} />
      <br />
    </div>
    <div>
      <textarea placeholder=\\"Enter a note for your Expense(optional)\\" value=\\"\\" onChange={[Function]} />
    </div>
    <div>
      <withStyles(SingleDatePicker) date={{...}} onDateChange={[Function]} focused={false} onFocusChange={[Function]} numberOfMonths={1} isOutsideRange={[Function: isOutsideRange]} id=\\"date\\" placeholder=\\"Date\\" ariaLabel={[undefined]} disabled={false} required={false} readOnly={false} screenReaderInputMessage=\\"\\" showClearDate={false} showDefaultInputIcon={false} inputIconPosition=\\"before\\" customInputIcon={{...}} customCloseIcon={{...}} noBorder={false} block={false} small={false} regular={false} verticalSpacing={22} keepFocusOnInput={false} orientation=\\"horizontal\\" anchorDirection=\\"left\\" openDirection=\\"down\\" horizontalMargin={0} withPortal={false} withFullScreenPortal={false} appendToBody={false} disableScroll={false} initialVisibleMonth={{...}} firstDayOfWeek={{...}} keepOpenOnDateSelect={false} reopenPickerOnClearDate={false} renderCalendarInfo={{...}} calendarInfoPosition=\\"bottom\\" hideKeyboardShortcutsPanel={false} daySize={39} isRTL={false} verticalHeight={{...}} transitionDuration={[undefined]} horizontalMonthPadding={13} navPrev={{...}} navNext={{...}} onPrevMonthClick={[Function: onPrevMonthClick]} onNextMonthClick={[Function: onNextMonthClick]} onClose={[Function: onClose]} renderMonthText={{...}} renderCalendarDay={[undefined]} renderDayContents={{...}} renderMonthElement={{...}} enableOutsideDays={false} isDayBlocked={[Function: isDayBlocked]} isDayHighlighted={[Function: isDayHighlighted]} displayFormat={[Function: displayFormat]} monthFormat=\\"MMMM YYYY\\" weekDayFormat=\\"dd\\" phrases={{...}} dayAriaLabelFormat={[undefined]} />
      <br />
    </div>
    <button>
      Add Expense
    </button>
  </form>
</div>"
`;

exports[` Should set note on textarea change 1`] = `
"<div>
  <h1>
    ExpenseForm
  </h1>
  <form onSubmit={[Function]}>
    <div>
      <input type=\\"text\\" placeholder=\\"AddExpense\\" value=\\"\\" onChange={[Function]} autoFocus={true} />
      <input type=\\"number\\" placeholder=\\"Enter Amount\\" value=\\"\\" onChange={[Function]} />
      <br />
    </div>
    <div>
      <textarea placeholder=\\"Enter a note for your Expense(optional)\\" value=\\"note\\" onChange={[Function]} />
    </div>
    <div>
      <withStyles(SingleDatePicker) date={{...}} onDateChange={[Function]} focused={false} onFocusChange={[Function]} numberOfMonths={1} isOutsideRange={[Function: isOutsideRange]} id=\\"date\\" placeholder=\\"Date\\" ariaLabel={[undefined]} disabled={false} required={false} readOnly={false} screenReaderInputMessage=\\"\\" showClearDate={false} showDefaultInputIcon={false} inputIconPosition=\\"before\\" customInputIcon={{...}} customCloseIcon={{...}} noBorder={false} block={false} small={false} regular={false} verticalSpacing={22} keepFocusOnInput={false} orientation=\\"horizontal\\" anchorDirection=\\"left\\" openDirection=\\"down\\" horizontalMargin={0} withPortal={false} withFullScreenPortal={false} appendToBody={false} disableScroll={false} initialVisibleMonth={{...}} firstDayOfWeek={{...}} keepOpenOnDateSelect={false} reopenPickerOnClearDate={false} renderCalendarInfo={{...}} calendarInfoPosition=\\"bottom\\" hideKeyboardShortcutsPanel={false} daySize={39} isRTL={false} verticalHeight={{...}} transitionDuration={[undefined]} horizontalMonthPadding={13} navPrev={{...}} navNext={{...}} onPrevMonthClick={[Function: onPrevMonthClick]} onNextMonthClick={[Function: onNextMonthClick]} onClose={[Function: onClose]} renderMonthText={{...}} renderCalendarDay={[undefined]} renderDayContents={{...}} renderMonthElement={{...}} enableOutsideDays={false} isDayBlocked={[Function: isDayBlocked]} isDayHighlighted={[Function: isDayHighlighted]} displayFormat={[Function: displayFormat]} monthFormat=\\"MMMM YYYY\\" weekDayFormat=\\"dd\\" phrases={{...}} dayAriaLabelFormat={[undefined]} />
      <br />
    </div>
    <button>
      Add Expense
    </button>
  </form>
</div>"
`;

exports[` Should set valid amount on input change 1`] = `
"<div>
  <h1>
    ExpenseForm
  </h1>
  <form onSubmit={[Function]}>
    <div>
      <input type=\\"text\\" placeholder=\\"AddExpense\\" value=\\"\\" onChange={[Function]} autoFocus={true} />
      <input type=\\"number\\" placeholder=\\"Enter Amount\\" value=\\"23.23\\" onChange={[Function]} />
      <br />
    </div>
    <div>
      <textarea placeholder=\\"Enter a note for your Expense(optional)\\" value=\\"\\" onChange={[Function]} />
    </div>
    <div>
      <withStyles(SingleDatePicker) date={{...}} onDateChange={[Function]} focused={false} onFocusChange={[Function]} numberOfMonths={1} isOutsideRange={[Function: isOutsideRange]} id=\\"date\\" placeholder=\\"Date\\" ariaLabel={[undefined]} disabled={false} required={false} readOnly={false} screenReaderInputMessage=\\"\\" showClearDate={false} showDefaultInputIcon={false} inputIconPosition=\\"before\\" customInputIcon={{...}} customCloseIcon={{...}} noBorder={false} block={false} small={false} regular={false} verticalSpacing={22} keepFocusOnInput={false} orientation=\\"horizontal\\" anchorDirection=\\"left\\" openDirection=\\"down\\" horizontalMargin={0} withPortal={false} withFullScreenPortal={false} appendToBody={false} disableScroll={false} initialVisibleMonth={{...}} firstDayOfWeek={{...}} keepOpenOnDateSelect={false} reopenPickerOnClearDate={false} renderCalendarInfo={{...}} calendarInfoPosition=\\"bottom\\" hideKeyboardShortcutsPanel={false} daySize={39} isRTL={false} verticalHeight={{...}} transitionDuration={[undefined]} horizontalMonthPadding={13} navPrev={{...}} navNext={{...}} onPrevMonthClick={[Function: onPrevMonthClick]} onNextMonthClick={[Function: onNextMonthClick]} onClose={[Function: onClose]} renderMonthText={{...}} renderCalendarDay={[undefined]} renderDayContents={{...}} renderMonthElement={{...}} enableOutsideDays={false} isDayBlocked={[Function: isDayBlocked]} isDayHighlighted={[Function: isDayHighlighted]} displayFormat={[Function: displayFormat]} monthFormat=\\"MMMM YYYY\\" weekDayFormat=\\"dd\\" phrases={{...}} dayAriaLabelFormat={[undefined]} />
      <br />
    </div>
    <button>
      Add Expense
    </button>
  </form>
</div>"
`;

exports[` error for invalid form submission 1`] = `
"<div>
  <h1>
    ExpenseForm
  </h1>
  <span style={{...}}>
    Please provide description and amount
  </span>
  <form onSubmit={[Function]}>
    <div>
      <input type=\\"text\\" placeholder=\\"AddExpense\\" value=\\"\\" onChange={[Function]} autoFocus={true} />
      <input type=\\"number\\" placeholder=\\"Enter Amount\\" value=\\"\\" onChange={[Function]} />
      <br />
    </div>
    <div>
      <textarea placeholder=\\"Enter a note for your Expense(optional)\\" value=\\"\\" onChange={[Function]} />
    </div>
    <div>
      <withStyles(SingleDatePicker) date={{...}} onDateChange={[Function]} focused={false} onFocusChange={[Function]} numberOfMonths={1} isOutsideRange={[Function: isOutsideRange]} id=\\"date\\" placeholder=\\"Date\\" ariaLabel={[undefined]} disabled={false} required={false} readOnly={false} screenReaderInputMessage=\\"\\" showClearDate={false} showDefaultInputIcon={false} inputIconPosition=\\"before\\" customInputIcon={{...}} customCloseIcon={{...}} noBorder={false} block={false} small={false} regular={false} verticalSpacing={22} keepFocusOnInput={false} orientation=\\"horizontal\\" anchorDirection=\\"left\\" openDirection=\\"down\\" horizontalMargin={0} withPortal={false} withFullScreenPortal={false} appendToBody={false} disableScroll={false} initialVisibleMonth={{...}} firstDayOfWeek={{...}} keepOpenOnDateSelect={false} reopenPickerOnClearDate={false} renderCalendarInfo={{...}} calendarInfoPosition=\\"bottom\\" hideKeyboardShortcutsPanel={false} daySize={39} isRTL={false} verticalHeight={{...}} transitionDuration={[undefined]} horizontalMonthPadding={13} navPrev={{...}} navNext={{...}} onPrevMonthClick={[Function: onPrevMonthClick]} onNextMonthClick={[Function: onNextMonthClick]} onClose={[Function: onClose]} renderMonthText={{...}} renderCalendarDay={[undefined]} renderDayContents={{...}} renderMonthElement={{...}} enableOutsideDays={false} isDayBlocked={[Function: isDayBlocked]} isDayHighlighted={[Function: isDayHighlighted]} displayFormat={[Function: displayFormat]} monthFormat=\\"MMMM YYYY\\" weekDayFormat=\\"dd\\" phrases={{...}} dayAriaLabelFormat={[undefined]} />
      <br />
    </div>
    <button>
      Add Expense
    </button>
  </form>
</div>"
`;

exports[`Should refer expensesList with expenses 1`] = `
<div>
  <h1>
    ExpenseForm
  </h1>
  
  <form
    onSubmit={[Function]}
  >
    <div>
      <input
        autoFocus={true}
        onChange={[Function]}
        placeholder="AddExpense"
        type="text"
        value="Coffee"
      />
      <input
        onChange={[Function]}
        placeholder="Enter Amount"
        type="number"
        value="9500"
      />
      <br />
    </div>
    <div>
      <textarea
        onChange={[Function]}
        placeholder="Enter a note for your Expense(optional)"
        value=""
      />
    </div>
    <div>
      <withStyles(SingleDatePicker)
        anchorDirection="left"
        appendToBody={false}
        block={false}
        calendarInfoPosition="bottom"
        customCloseIcon={null}
        customInputIcon={null}
        date={"1970-01-01T00:00:00.400Z"}
        daySize={39}
        disableScroll={false}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        firstDayOfWeek={null}
        focused={false}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        horizontalMonthPadding={13}
        id="date"
        initialVisibleMonth={null}
        inputIconPosition="before"
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepFocusOnInput={false}
        keepOpenOnDateSelect={false}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        noBorder={false}
        numberOfMonths={1}
        onClose={[Function]}
        onDateChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        openDirection="down"
        orientation="horizontal"
        phrases={
          Object {
            "calendarLabel": "Calendar",
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsSelected": [Function],
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month.",
            "jumpToPrevMonth": "Move backward to switch to the previous month.",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
            "moveFocusByOneMonth": "Switch months.",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
            "openThisPanel": "Open this panel.",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field.",
            "selectFocusedDate": "Select the date in focus.",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        placeholder="Date"
        readOnly={false}
        regular={false}
        renderCalendarInfo={null}
        renderDayContents={null}
        renderMonthElement={null}
        renderMonthText={null}
        reopenPickerOnClearDate={false}
        required={false}
        screenReaderInputMessage=""
        showClearDate={false}
        showDefaultInputIcon={false}
        small={false}
        verticalHeight={null}
        verticalSpacing={22}
        weekDayFormat="dd"
        withFullScreenPortal={false}
        withPortal={false}
      />
      <br />
    </div>
    <button>
      Add Expense
    </button>
  </form>
</div>
`;

exports[`Should refer expensesList without expenses 1`] = `
<div>
  <h1>
    ExpenseForm
  </h1>
  
  <form
    onSubmit={[Function]}
  >
    <div>
      <input
        autoFocus={true}
        onChange={[Function]}
        placeholder="AddExpense"
        type="text"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="Enter Amount"
        type="number"
        value=""
      />
      <br />
    </div>
    <div>
      <textarea
        onChange={[Function]}
        placeholder="Enter a note for your Expense(optional)"
        value=""
      />
    </div>
    <div>
      <withStyles(SingleDatePicker)
        anchorDirection="left"
        appendToBody={false}
        block={false}
        calendarInfoPosition="bottom"
        customCloseIcon={null}
        customInputIcon={null}
        date={"1970-01-01T00:00:00.000Z"}
        daySize={39}
        disableScroll={false}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        firstDayOfWeek={null}
        focused={false}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        horizontalMonthPadding={13}
        id="date"
        initialVisibleMonth={null}
        inputIconPosition="before"
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepFocusOnInput={false}
        keepOpenOnDateSelect={false}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        noBorder={false}
        numberOfMonths={1}
        onClose={[Function]}
        onDateChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        openDirection="down"
        orientation="horizontal"
        phrases={
          Object {
            "calendarLabel": "Calendar",
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsSelected": [Function],
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month.",
            "jumpToPrevMonth": "Move backward to switch to the previous month.",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
            "moveFocusByOneMonth": "Switch months.",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
            "openThisPanel": "Open this panel.",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field.",
            "selectFocusedDate": "Select the date in focus.",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        placeholder="Date"
        readOnly={false}
        regular={false}
        renderCalendarInfo={null}
        renderDayContents={null}
        renderMonthElement={null}
        renderMonthText={null}
        reopenPickerOnClearDate={false}
        required={false}
        screenReaderInputMessage=""
        showClearDate={false}
        showDefaultInputIcon={false}
        small={false}
        verticalHeight={null}
        verticalSpacing={22}
        weekDayFormat="dd"
        withFullScreenPortal={false}
        withPortal={false}
      />
      <br />
    </div>
    <button>
      Add Expense
    </button>
  </form>
</div>
`;

exports[`Should test submit with error and with complete info 1`] = `
"<div>
  <h1>
    ExpenseForm
  </h1>
  <form onSubmit={[Function]}>
    <div>
      <input type=\\"text\\" placeholder=\\"AddExpense\\" value=\\"Coffee\\" onChange={[Function]} autoFocus={true} />
      <input type=\\"number\\" placeholder=\\"Enter Amount\\" value=\\"9500\\" onChange={[Function]} />
      <br />
    </div>
    <div>
      <textarea placeholder=\\"Enter a note for your Expense(optional)\\" value=\\"\\" onChange={[Function]} />
    </div>
    <div>
      <withStyles(SingleDatePicker) date={{...}} onDateChange={[Function]} focused={false} onFocusChange={[Function]} numberOfMonths={1} isOutsideRange={[Function: isOutsideRange]} id=\\"date\\" placeholder=\\"Date\\" ariaLabel={[undefined]} disabled={false} required={false} readOnly={false} screenReaderInputMessage=\\"\\" showClearDate={false} showDefaultInputIcon={false} inputIconPosition=\\"before\\" customInputIcon={{...}} customCloseIcon={{...}} noBorder={false} block={false} small={false} regular={false} verticalSpacing={22} keepFocusOnInput={false} orientation=\\"horizontal\\" anchorDirection=\\"left\\" openDirection=\\"down\\" horizontalMargin={0} withPortal={false} withFullScreenPortal={false} appendToBody={false} disableScroll={false} initialVisibleMonth={{...}} firstDayOfWeek={{...}} keepOpenOnDateSelect={false} reopenPickerOnClearDate={false} renderCalendarInfo={{...}} calendarInfoPosition=\\"bottom\\" hideKeyboardShortcutsPanel={false} daySize={39} isRTL={false} verticalHeight={{...}} transitionDuration={[undefined]} horizontalMonthPadding={13} navPrev={{...}} navNext={{...}} onPrevMonthClick={[Function: onPrevMonthClick]} onNextMonthClick={[Function: onNextMonthClick]} onClose={[Function: onClose]} renderMonthText={{...}} renderCalendarDay={[undefined]} renderDayContents={{...}} renderMonthElement={{...}} enableOutsideDays={false} isDayBlocked={[Function: isDayBlocked]} isDayHighlighted={[Function: isDayHighlighted]} displayFormat={[Function: displayFormat]} monthFormat=\\"MMMM YYYY\\" weekDayFormat=\\"dd\\" phrases={{...}} dayAriaLabelFormat={[undefined]} />
      <br />
    </div>
    <button>
      Add Expense
    </button>
  </form>
</div>"
`;

exports[`should not display if amount is invalid on input change 1`] = `
"<div>
  <h1>
    ExpenseForm
  </h1>
  <form onSubmit={[Function]}>
    <div>
      <input type=\\"text\\" placeholder=\\"AddExpense\\" value=\\"\\" onChange={[Function]} autoFocus={true} />
      <input type=\\"number\\" placeholder=\\"Enter Amount\\" value=\\"\\" onChange={[Function]} />
      <br />
    </div>
    <div>
      <textarea placeholder=\\"Enter a note for your Expense(optional)\\" value=\\"\\" onChange={[Function]} />
    </div>
    <div>
      <withStyles(SingleDatePicker) date={{...}} onDateChange={[Function]} focused={false} onFocusChange={[Function]} numberOfMonths={1} isOutsideRange={[Function: isOutsideRange]} id=\\"date\\" placeholder=\\"Date\\" ariaLabel={[undefined]} disabled={false} required={false} readOnly={false} screenReaderInputMessage=\\"\\" showClearDate={false} showDefaultInputIcon={false} inputIconPosition=\\"before\\" customInputIcon={{...}} customCloseIcon={{...}} noBorder={false} block={false} small={false} regular={false} verticalSpacing={22} keepFocusOnInput={false} orientation=\\"horizontal\\" anchorDirection=\\"left\\" openDirection=\\"down\\" horizontalMargin={0} withPortal={false} withFullScreenPortal={false} appendToBody={false} disableScroll={false} initialVisibleMonth={{...}} firstDayOfWeek={{...}} keepOpenOnDateSelect={false} reopenPickerOnClearDate={false} renderCalendarInfo={{...}} calendarInfoPosition=\\"bottom\\" hideKeyboardShortcutsPanel={false} daySize={39} isRTL={false} verticalHeight={{...}} transitionDuration={[undefined]} horizontalMonthPadding={13} navPrev={{...}} navNext={{...}} onPrevMonthClick={[Function: onPrevMonthClick]} onNextMonthClick={[Function: onNextMonthClick]} onClose={[Function: onClose]} renderMonthText={{...}} renderCalendarDay={[undefined]} renderDayContents={{...}} renderMonthElement={{...}} enableOutsideDays={false} isDayBlocked={[Function: isDayBlocked]} isDayHighlighted={[Function: isDayHighlighted]} displayFormat={[Function: displayFormat]} monthFormat=\\"MMMM YYYY\\" weekDayFormat=\\"dd\\" phrases={{...}} dayAriaLabelFormat={[undefined]} />
      <br />
    </div>
    <button>
      Add Expense
    </button>
  </form>
</div>"
`;
